boards/common/nucleo32/include/arduino_pinmap.h

```diff
@@ -39,7 +39,7 @@ extern "C" {
 #define ARDUINO_PIN_4           GPIO_PIN(PORT_B, 7)
 #define ARDUINO_PIN_5           GPIO_PIN(PORT_B, 6)
 #define ARDUINO_PIN_6           GPIO_PIN(PORT_B, 1)
-#if defined(CPU_MODEL_STM32L031K6) || defined(CPU_MODEL_STM32L432KC)
+#ifndef GPIOF
```

Comment:
IMHO, the use of `PORT_F` for this pins doesn't depend on the fact that it exists or not. This is something that is board level related only. So even if it's less maintainable, we need to have an explicit list of boards here.
Just adding nucleo-l412kb to this list will be ok.

Reply:
> So even if it's less maintainable, we need to have an explicit list of boards here.

OK. What about NUCLEO-L011K4. Should we add also the check for `CPU_MODEL_STM32412KM` although we don't have a board definition for it yet.

Reply:
> What about NUCLEO-L011K4

It is not supported by RIOT, too small. I think we can ignore boards that are not supported yet.

Reply:
Fixed.

I have also added a check in `gpio_init` which leads to a failed assertion if the given port is not available. I added it because I went into the problem in `tests/periph_gpio` that RIOT simply crashes if port is used that is not available. Now you get a failed assertion. The check is only executed if `NDEBUG` is not set.

We can discard the check if you think that a port that is not available should not be used. As long as named ports are used, the compiler will already do the job.

Reply:
> I have also added a check in gpio_init which leads to a failed assertion if the given port is not available. 

I'm not sure this is that useful. As you noticed, the compiler will already perform that check anyway. In `tests/periph_gpio` the user is allowed to use a port that doesn't exist but since this is a test application with a shell, I don't think this is super critical.

Maybe open a PR for that specific issue so we can merge the initial cleanup ?